client/src/pages/Signin/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+
+import AuthForm from "components/AuthForm";
+import { signInUser } from "app/userSlice";
+import { removeError } from "app/errorSlice";
+
+export default function SignUp() {
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+    const { isAuthenticated } = useSelector((state) => state.user);
+    const { message: error } = useSelector((state) => state.error);
+    const [submitted, setSubmitted] = useState(false);
+
+    useEffect(() => {
+        dispatch(removeError());
+    }, []);
+
+    // Show error if error !== null
+    useEffect(() => {
+        if (isAuthenticated) navigate("/");
```

Comment:
do you still need to update code here? I am a little confused about this `useEffect(() => {
        if (isAuthenticated) navigate("/");`

Reply:
> do you still need to update code here? I am a little confused about this `useEffect(() => { if (isAuthenticated) navigate("/");`

Sorry about the comment here. I forgot to delete that comment. This is for auto navigation for browsing the signin page when someone has logged. The logic is: if you want to sign in without sign out, you have to sign out first, or it will redirect to the home page.